### ISSUE TITLE
Segmentation fault using ma.masked_all with Python 3.3

### ISSUE THREAD

**Issue by taldcroft** [numpy/numpy#3276]
Using `ma.masked_all()` with a unicode string on linux-64 with python 3.3 and numpy 1.7.0, there is a segmentation fault shown below.  This is not present in python 3.2 or python 2.7.

```
ccosmos$ ipython3
Python 3.3.0 |Anaconda 1.4.0 (64-bit)| (default, Feb 25 2013, 15:57:27) 
...

In [1]: import numpy.ma as ma

In [2]: t = ma.masked_all(3, dtype=[('a', '<U3')])

In [3]: print(t)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-d6c4f5caf31e> in <module>()
----> 1 print(t)

/home/aldcroft/soft/anaconda/envs/py3/lib/python3.3/site-packages/numpy/ma/core.py in __str__(self)
   3538                 else:
   3539                     rdtype = _recursive_make_descr(self.dtype, "O")
-> 3540                     res = self._data.astype(rdtype)
   3541                     _recursive_printoption(res, m, f)
   3542         else:

ValueError: character U+53000464 is not in range [U+0000; U+10ffff]

In [4]: print(repr(t))
Segmentation fault
```

**Comment by jjhelmus** [comment]
This bug was fixed by 007b186a853c8ab37db62042376a9b2d66c18dcf, part of PR #4997 and can be closed.  


``` Python
In [2]: import numpy.ma as ma

In [3]: t = ma.masked_all(3, dtype=[('a', '<U3')])

In [4]: print(t)
[(--,) (--,) (--,)]

In [5]: print(repr(t))
masked_array(data = [(--,) (--,) (--,)],
             mask = [(True,) (True,) (True,)],
       fill_value = ('N/A',),
            dtype = [('a', '<U3')])
```

**Comment by charris** [comment]
@jjhelmus Thanks for the heads up.
